enclave/core/hostcalls.c

```diff
@@ -13,6 +13,40 @@
 
 #include "core_t.h"
 
+#if !defined(OE_USE_BUILTIN_EDL)
+/**
+ * Declare the protoype of the following functions to avoid the
+ * missing-prototypes warning.
+ */
+oe_result_t _oe_log_ocall(uint32_t log_level, const char* message);
+
+oe_result_t _oe_write_ocall(int device, const char* str, size_t maxlen);
+
+/**
+ * Make the following OCALLs weak to support the system EDL opt-in.
+ * When the user does not opt in (import) the EDL, the linker will picked
+ * the following default implementations. If the user opts in the EDL,
+ * the implementions (which are strong) in the oeedger8r-generated code will be
+ * used.
+ */
+oe_result_t _oe_log_ocall(uint32_t log_level, const char* message)
+{
+    OE_UNUSED(log_level);
+    OE_UNUSED(message);
+    return OE_UNSUPPORTED;
+}
+OE_WEAK_ALIAS(_oe_log_ocall, oe_log_ocall);
```

Comment:
Is WEAK_ALIAS needed? The user has to pass liboe_logging_host.a or equivalent library to the linker line. Therefore those definitions would get picked up.
Can these stubs be entirely removed?

Reply:
What do you mean by liboe_logging_host.a? I may missed something. Are we going to create a new library?

Reply:
@anakrish this pr is just to allow users to opt out of logging altogether. It doesn’t attempt to allow users to swap in their own logging framework (although you could technically do that with a similar mechanism)

Reply:
@anakrish this pr is just to allow users to opt out of logging altogether. It doesn’t attempt to allow users to swap in their own logging framework (although you could technically do that with a similar mechanism)